docs: describe CRUD spec compliance (#2944)
mongodb/node-mongodb-native: docs/CHANGES_4.0.0.md

If that's outside your support matrix at this time, that's okay!
Bug fix support for our 3.x branch will not be ending until summer 2022, which has support going back as far as Node.js v4!

### CRUD results

> Our CRUD operations now return the driver's wide spec compliant results which can be viewed here:

Reviewer: ```suggestion
Our CRUD operations now return the drivers wide spec compliant results which are defined here:
```

Reviewer 2: I think hyphenating would make this more readable, "drivers-wide spec-compliant results"
